### PR TITLE
Adds diagnosis to the NewCrop SOAP message

### PR DESCRIPTION
**Pull request by kchapple**
Adds diagnosis to the NewCrop SOAP message so diagnosis appears in the drug signature list (now required by medicare) and some bug fixes

Fixes #

### CODE REVIEW

File: interface/eRxXMLBuilder.php [hunk 1]
```diff
@@ -711,6 +711,41 @@ public function getPatientFreeformAllergy($patientId)
...
+    {
+        $diagnosisData = $this->getStore()
+            ->getPatientDiagnosisByPatientId($patientId);
+
+        $elements = array();
+
+        while ($diagnosis = sqlFetchArray($diagnosisData)) {
+            $element = $this->getDocument()->createElement('PatientDiagnosis');
+
+            if ($diagnosis['diagnosis']) {
```
[the remark this answers is not in the document]
Comment (bradymiller) [reply interface/eRxXMLBuilder.php:724]: check out below screenshots for workflow that creates that format (last screenshot is how it looks in database)

image

image

image

image

image

[the remark this answers is not in the document]
Comment (kchapple) [reply interface/eRxXMLBuilder.php:724]: Thanks @bradymiller Very helpful. The fix for multiple has been pushed.

**Comment by kchapple** [comment]
Not sure why build failed. Looks like a permissions error on the build machine?

**Comment by stephenwaite** [comment]
hi @kchapple , we get random fails, no need to investigate unless you want to roll up your sleeves with our github actions :)

**Comment by stephenwaite** [comment]
thanks for fixing up up the diags, will bring in

**Comment by kchapple** [comment]
@bradymiller @stephenwaite if possible can this get put in next patch? It contains MySQL 8 fixes that are breaking Newcrop new 600.3 installations. Thanks!

**Comment by stephenwaite** [comment]
just added to [rel-600](https://github.com/openemr/openemr/commit/1259a2b13ba5a3e7c6e96be9b73fdac2751477dc) so will get in next patch which should be out pretty soon

**Pull request merged**